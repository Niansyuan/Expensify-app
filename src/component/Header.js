import React from "react";
import { Link } from 'react-router-dom';
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="contain-container">
            <div className="header__contain">
                <Link
                    to='/dashboard'
                    className="header__title"
                >
                    <h1>Expensify</h1>
                </Link>
                <button
                    className="button--link"
                    onClick={startLogout}
                ><i className="fas fa-sign-out-alt"></i>Logout</button>
            </div>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);