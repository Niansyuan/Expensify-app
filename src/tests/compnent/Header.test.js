//react-test-renderer
import React from "react";
import Header from "../../component/Header";
import { shallow } from "enzyme";
import "../setupTests";

// import ShallowRenderer from "react-test-renderer/shallow";
//test with react-test-renderer/shallow
// test('should render Header correctly',()=>{
//     const renderer=new ShallowRenderer();
//     renderer.render(<Header />);
//     expect(renderer.getRenderOutput()).toMatchSnapshot();
// });

//test with enzyme
test('should render Header correctly',()=>{
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
})