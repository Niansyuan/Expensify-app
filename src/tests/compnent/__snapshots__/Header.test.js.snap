// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="contain-container"
  >
    <div
      className="header__contain"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <button
        className="button--link"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        Logout
      </button>
    </div>
  </div>
</header>
`;
